src/main/java/ch/njol/skript/sections/SecOpenInventory.java

```diff
@@ -0,0 +1,150 @@
+		private final Inventory inventory;
+
+		public InventorySectionEvent(Inventory inventory) {
+			this.inventory = inventory;
+		}
+
+		public Inventory getInventory() {
+			return inventory;
+		}
+
+		@Override
+		@NotNull
+		public HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+	}
+
+	static {
+		Skript.registerSection(SecOpenInventory.class, "[(show|open|create)] %inventory/inventorytype%");
+	}
+
+	@Nullable
+	private Expression<?> inventory;
+
+	@Nullable
+	private Expression<Player> players;
+
+	@Nullable
+	private Trigger trigger;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult,
+			SectionNode sectionNode, List<TriggerItem> triggerItems) {
+
+		inventory = exprs[0];
+		players = (Expression<Player>) exprs[1];
+		if (inventory instanceof Literal) {
+			Literal<?> literal = (Literal<?>) inventory;
+			Object object = literal.getSingle();
+			if (object instanceof InventoryType && !((InventoryType) object).isCreatable()) {
+				Skript.error("You can't open a '" + literal.toString() + "' inventory to players. It's not creatable.");
+				return false;
+			}
+		}
+		trigger = loadCode(sectionNode, "open inventory", InventorySectionEvent.class);
+		return true;
+	}
+
+	@Override
+	protected TriggerItem walk(Event event) {
+		Inventory inventory = null;
+		Object object = this.inventory.getSingle(event);
+		if (object == null)
+			return super.walk(event, false);
+		if (object instanceof Inventory) {
+			inventory = (Inventory) object;
+		} else if (object instanceof InventoryType) {
+			try {
+				inventory = Bukkit.createInventory(null, (InventoryType) object);
+			} catch (Exception e) {
+				// Spigot forgot to label some InventoryType's as non creatable in some versions < 1.19.4
+				// So this throws NullPointerException aswell ontop of the IllegalArgumentException.
+				// See https://hub.spigotmc.org/jira/browse/SPIGOT-7301
+				Skript.error("You can't open a '" + Classes.toString((InventoryType) object) + "' inventory to players. It's not creatable.");
+			}
+		} else {
+			assert false;
+		}
+		if (inventory == null)
+			return super.walk(event, false);
+		InventorySectionEvent inventoryEvent = new InventorySectionEvent(inventory);
+		Object localVars = Variables.copyLocalVariables(event);
+		Variables.setLocalVariables(inventoryEvent, localVars);
+		TriggerItem.walk(trigger, inventoryEvent);
+		Variables.setLocalVariables(event, Variables.copyLocalVariables(inventoryEvent));
+		Variables.removeLocals(inventoryEvent);
+		if (players != null)
+			for (Player player : players.getArray(event))
+				player.openInventory(inventory);
+		return super.walk(event, false);
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "section open " + inventory.toString(event, debug) + " to " + players.toString(event, debug);
```

Comment:
```suggestion
		return "open " + inventory.toString(event, debug) + " to " + players.toString(event, debug);
```
I know syntax is same/similar as the effect but we should still represent the "actual" string of the syntax